src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/BaseTest.java

```diff
@@ -1,71 +1,77 @@
-import liquibase.Contexts;
-import liquibase.Liquibase;
-import liquibase.database.Database;
-import liquibase.database.DatabaseFactory;
-import liquibase.database.jvm.JdbcConnection;
-import liquibase.resource.ClassLoaderResourceAccessor;
+import org.flywaydb.core.Flyway;
 import org.junit.BeforeClass;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Primary;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
 
 @Configuration
-public class BaseTest {
+public abstract class BaseTest {
 
     protected static final ObjectMapper mapper = new ObjectMapper();
-    public static final String POSTGRES = "postgres";
 
     @BeforeClass
     public static void init() {
         mapper.registerModule(new JavaTimeModule());
         mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
     }
 
-    @Bean
-    @Primary
-    public EmbeddedPostgres embeddedPostgres() throws IOException {
-        return EmbeddedPostgres
-            .builder()
-            .setConnectConfig("stringtype", "unspecified")
-            .setPort(0)
-            .start();
-    }
+    protected static final MediaType JSON_CONTENT_TYPE = new MediaType(
+            MediaType.APPLICATION_JSON.getType(),
+            MediaType.APPLICATION_JSON.getSubtype(),
+            StandardCharsets.UTF_8
+    );
+
+    @TestConfiguration
+    static class Configuration {
+        Connection connection;
+
+        @Bean
+        public EmbeddedPostgres embeddedPostgres() throws IOException {
+            return EmbeddedPostgres
+                    .builder()
+                    .setPort(0)
+                    .start();
+        }
 
-    @Bean
-    @Primary
-    public DataSource dataSource(@Qualifier("embeddedPostgres") final EmbeddedPostgres pg) throws Exception {
+        @Bean
+        public DataSource dataSource(@Autowired EmbeddedPostgres pg) throws Exception {
 
-        DataSource datasource = DataSourceBuilder
-            .create()
-            .username(POSTGRES)
-            .password(POSTGRES)
-            .url(pg.getJdbcUrl(POSTGRES, POSTGRES))
-            .driverClassName("org.postgresql.Driver")
-            .build();
+            final Properties props = new Properties();
+            // Instruct JDBC to accept JSON string for JSONB
+            props.setProperty("stringtype", "unspecified");
+            connection = DriverManager.getConnection(pg.getJdbcUrl("postgres", "postgres"), props);
+            DataSource datasource = new SingleConnectionDataSource(connection, true);
+            Flyway.configure().dataSource(datasource)
```

Comment:
do we need this configuration as we already defined in application.yaml?

Reply:
Each of the integration tests fail when this config is not included here, I messed around with it a bit and couldn't see why it doesn't work without it.